### ISSUE TITLE
WW3/OCN Required Coupling Fields (export/import)

### ISSUE THREAD

**Issue by pvelissariou1** [oceanmodeling/ufs-coastal#32]
## Description

Discussion on the required import/export fields for 2D/3D coupling between WW3 and OCN

## Suggestion

Please check the suggested requirements and add/modify as needed the following google document: [ww3-exports-ocn-3Dwave-terms](https://docs.google.com/document/d/140_xAMOb30iljGL09C_S7tNAKGDY2j8GEjPb7Pg3Xxg/edit?usp=drive_link)

**Comment by josephzhang8** [comment]
It's easy for me to change the names on SCHISM side.

**Comment by pvelissariou1** [comment]
@josephzhang8 Thank you, let's see how the WW3 folks will respond. Could you please comment on the export fields from WW3 I listed in the document. I believe the list is complete and it covers all wave induced terms in the OCN side.

**Comment by josephzhang8** [comment]
@pvelissariou1 
RS* are 2D radiation stress from Longuet-Higgins & Stewart formulation.
There will be many more once we change to vortex formulation.

**Comment by gseroka** [comment]
Thanks @pvelissariou1 and all. 

> At the request of @uturuncoglu I have added additional outputs from export from WW3 via UFS so he can check them.

@josephzhang8 did you add these to the Google [document](https://docs.google.com/document/d/140_xAMOb30iljGL09C_S7tNAKGDY2j8GEjPb7Pg3Xxg/edit)?

> @uturuncoglu changed the exported radiation stress field names in WW3 to work in UFS-Coastal.

> @pvelissariou1 RS* are 2D radiation stress from Longuet-Higgins & Stewart formulation. There will be many more once we change to vortex formulation.

@pvelissariou1 are the radiation stresses (RS* or other) included in the Google [document](https://docs.google.com/document/d/140_xAMOb30iljGL09C_S7tNAKGDY2j8GEjPb7Pg3Xxg/edit)?

**Comment by josephzhang8** [comment]
Thx @gseroka! I did not know that doc but will review/edit.

**Comment by pvelissariou1** [comment]
@gseroka Greg, the RS* fields are not the fields exported by WW3/NUOPC, they are SCHISM's internal definitions. We pass the fields (described in the doc) via the ESMF/NUOPC api, for example as eastward_wave_radiation_stress for RSXX stress. The variables for the export fields in the doc need to be discussed.

**Comment by pvelissariou1** [comment]
@gseroka after your meeting today, please update the doc so I can finalize it for our future development tasks in UFS-Coastal. We are trying to document and organize our issues/tasks in UFS-Coastal the best we can, @janahaddad is leading these efforts (thank you so much Jana). Most likely, as we deal with many different components we will have some kind of documentation for all components. Also we need to keep track on the issues as often after model updates things break on the model component side (e.g., SCHISM the last days). @janahaddad, Jana we might start thinking of documenting what are the "stable" commits of each model component to make sure that UFS-Coastal compiles and runs its regression tests properly. Any advice/suggestions on these?

**Comment by janahaddad** [comment]
Hey @pvelissariou1, yes I've been thinking about this, and @uturuncoglu has a preferred method that's he's been moving forward with: the idea is to keep track of component-level issues in forks for that model component. For example [Ufuk recently created this ROMS fork for this purpose](https://github.com/oceanmodeling/roms/issues), and he's recently been working on WW3-level issues on https://github.com/oceanmodeling/ww3/issues 

Re: documenting status -- Ufuk has also recently updated the wiki page to track which configurations are working and passing RT's. Let's chat briefly about this at Monday's tagup!

**Comment by uturuncoglu** [comment]
@pvelissariou1 @janahaddad Please  just keep in your mind that WW3 has two NUOPC/Cap (old one and the NCAR one) and I think if we need to introduce new fields that needs to be done in new cap (also called mesh cap) since almost all the WW3 applications under UFS is using new cap.

**Comment by uturuncoglu** [comment]
@janahaddad Yes. It is better to open an issue in component side (in out forks). At this point, I test all the configurations (marked as bold) in the wiki page and all are working except ROMS one that I am looking at this point. I also need to update the wiki page and remove issue links from WW3 configurations. I'll do it soon.

**Comment by pvelissariou1** [comment]
@uturuncoglu We need to keep track on woking/stable commits of the model components so we can freeze the model components to a specific working commit. That is all I am saying. Let's talk about all these next week.

**Comment by uturuncoglu** [comment]
@pvelissariou1 Yes. That is my aim. Once ROMS issue is solved. I'll freeze the code in both model and component level and put restriction for direct push. Then, all updates and developments needs to pass those tests and create PR. We are very close.

**Comment by pvelissariou1** [comment]
@Joannes.Westerink.1@nd.edu, dwirasae@nd.edu, @jqi@umassd.edu, @c1chen@umassd.edu, @arango@marine.rutgers.edu, @gseroka, @saeed-moghimi-noaa, @ayumi.fujisaki@noaa.gov, @ayumif@umich.edu

Dear all,
Here at CMMB we have compiled a document the describes the requirements to couple WW3 with OCN for both 2D and 3D configurations. The document was shared with the EMC/WW3 developers for an initial evaluation/approval (1st iteration).
At your convinience, could you please review the document and add your suggestions/modifications? After all inputs have been incorporated, I will finalize the document to be discussed again with the WW3 developers for the next steps.
Here is the link to the document: [ww3-exports-ocn-3Dwave-terms](https://docs.google.com/document/d/140_xAMOb30iljGL09C_S7tNAKGDY2j8GEjPb7Pg3Xxg/edit?usp=drive_link)

Thank you so much for your help on this matter.
Takis

**Comment by josephzhang8** [comment]
Thank you @aronroland for your willingness to help. Can someone share the WW3 version with Aron used in UFS so he can take a look? Thx.

**Comment by saeed-moghimi-noaa** [comment]
@yunfangsun Would you please help @aronroland? Thanks

**Comment by pvelissariou1** [comment]
**_update on 04/03/2024_**

- - Need to follow with the various model developers to get their inputs (especially ROMS and FVCOM)
- At this point, I believe the document contains all necessary fields that need to be exported from WW3. I will check again and verify if anything else is missing (e.g. significant wave height, ...).
- @josephzhang8  will update the doc as needed following his incorporation of the vortex force into SCHISM.
-

**Comment by josephzhang8** [comment]
@pvelissariou1 : we are still waiting for you to tell us if the latest changes by Ufuk are good with your tests. I'd like to merge them to our master soon. Thx.

**Comment by pvelissariou1** [comment]
I am creating a few more test cases and testing right now. Overall I saw no issues so far. I will update in the next couple days.

**Comment by josephzhang8** [comment]
@uturuncoglu @pvelissariou1 

I just sent you a request to look into WW3 variables needed by SCHISM for 3D vortex formulation. I'll update g-doc now. Thx.

**Comment by josephzhang8** [comment]
@uturuncoglu : any updates on 3D vortex coupling? I can test it once the coding is complete. Thx.

**Comment by uturuncoglu** [comment]
@josephzhang8 Sorry. I could not make any progress on it. I am in a whole day meeting next two days but it would be nice to setup call to discuss about it.

**Comment by janahaddad** [comment]
@josephzhang8 would you be available in the next week for a 30-min meeting to strategize with us on how to move forward on this?

**Comment by josephzhang8** [comment]
Sure. We have a meeting 4pm on Monday May 13?

**Comment by josephzhang8** [comment]
I'll be travelling this week @ EPA workshop until Friday.




-Joseph



Joseph Zhang

Office: (804) 684 7466

Web: schism.wiki



________________________________

**Comment by janahaddad** [comment]
Ok thanks, let's plan to discuss at the existing meeting on Monday May 13! 

cc @saeed-moghimi-noaa
